src/Files.App/ServicesImplementation/QuickLookPreviewPopupService.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2023 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+using Files.Backend.Services;
+using Microsoft.Extensions.Logging;
+using System.IO;
+using System.IO.Pipes;
+using System.Security.Principal;
+
+namespace Files.App.ServicesImplementation
+{
+	/// <inheritdoc cref="IPreviewPopupService"/>
+	internal sealed class QuickLookPreviewPopupService : ObservableObject, IPreviewPopupService
+	{
+		private const int TIMEOUT = 500;
+		private static string pipeName = $"QuickLook.App.Pipe.{WindowsIdentity.GetCurrent().User?.Value}";
+		private static string pipeMessageSwitch = "QuickLook.App.PipeMessages.Switch";
+		private static string pipeMessageToggle = "QuickLook.App.PipeMessages.Toggle";
+
+		public async Task TogglePreviewPopup(string path)
```

Comment:
I think duplicate code can be reduced if there is a private method that has the message parameter and call it from the public API methods.